@here/harp-mapview-decoder/test/TileGeometryLoaderTest.ts

```diff
@@ -380,22 +343,27 @@ describe("TileGeometryLoader", function() {
 
             tile.dispose();
 
-            geometryLoader.waitFinished().should.be.rejected.then(() => {
-                expect(spyProcessTechniques.callCount).equal(1);
-                expect(spyCreateGeometries.callCount).equal(0, "should not create geometry");
-                expect(geometryLoader.isFinished).to.be.false;
+            // Dispose should have rejected the promise.
+            await geometryLoader.waitFinished().should.be.rejected;
 
-                geometryLoader.reset();
+            // Wait for the geometry creation task to return without creating any geometry.
+            expect(mapView.taskQueue.processNext(TileTaskGroups.CREATE)).to.be.true;
 
-                expect(mapView.taskQueue.numItemsLeft(TileTaskGroups.CREATE)).equal(1);
-                expect(mapView.taskQueue.processNext(TileTaskGroups.CREATE)).equal(true);
+            expect(spyProcessTechniques.callCount).equal(1);
+            expect(spyCreateGeometries.callCount).equal(0, "should not create geometry");
```

Comment:
I debugged this and they are incremented after the `geometryLoader!.update();` call, so they should be put up there.

Reply:
will do